src/bit-systems/network-debug.ts

```diff
@@ -0,0 +1,47 @@
+import { defineQuery, enterQuery, exitQuery, removeEntity } from "bitecs";
+import { Material, Scene, Vector3 } from "three";
+import { Text as TroikaText } from "troika-three-text";
+import { HubsWorld } from "../app";
+import { Networked } from "../bit-components";
+import { NetworkDebugPrefab } from "../prefabs/network-debug";
+import { formatComponentProps, formatObjectName } from "../react-components/debug-panel/ECSSidebar";
+import { renderAsEntity } from "../utils/jsx-entity";
+import { forEachMaterial } from "../utils/material-utils";
+import { EntityID } from "../utils/networking-types";
+
+const texts = new Map<EntityID, TroikaText>();
+const networkedQuery = defineQuery([Networked]);
+const enteredNetworkedQuery = enterQuery(networkedQuery);
+const exitedNetworkedQuery = exitQuery(networkedQuery);
+const _scale = new Vector3();
+export function networkDebugSystem(world: HubsWorld, scene: Scene) {
+  enteredNetworkedQuery(world).forEach(eid => {
+    const obj = world.eid2obj.get(eid);
+    if (obj) {
+      const textObj = world.eid2obj.get(renderAsEntity(world, NetworkDebugPrefab()))! as TroikaText;
+      texts.set(eid, textObj);
+      scene.add(textObj);
+      textObj.renderOrder = 999;
+      forEachMaterial(textObj, function (mat: Material) {
+        mat.depthTest = false;
+      });
+    }
+  });
+
+  exitedNetworkedQuery(world).forEach(eid => {
+    removeEntity(world, texts.get(eid)!.eid!);
+    texts.delete(eid);
```

Comment:
Don't we need to remove `textObj` from `scene`?

Reply:
Removing the object3D associated with an entity from the scene is handled by another system: 

https://github.com/mozilla/hubs/blob/d8e5400f8b123614f5e38dd5dd76d4ca05f47060/src/systems/remove-object3D-system.js#L100